### PR TITLE
[move] Update command line type parsing

### PR DESCRIPTION
**Pull request by tzakian**
## Description 

Update the move-command-line type parser to be a bit more general.

## Test plan 

Added additional tests + proptests.

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: external-crates/move/crates/move-command-line-common/src/parser.rs [hunk 1]
```diff
@@ -631,6 +633,10 @@ mod tests {
             "vector<0x1::M_::S_>",
             "vector<vector<0x1::M_::S_>>",
             "0x1::M::S<vector<u8>>",
+            "0x1::_bar::_BAR",
+            "0x1::__::__",
+            "0x1::_bar::_BAR<0x2::_____::______fooo______>",
+            "0x1::__::__<0x2::_____::______fooo______, 0xff::Bar____::_______foo>",
```
Comment (dariorussi) [inline external-crates/move/crates/move-command-line-common/src/parser.rs:639]: calm down now....

File: external-crates/move/crates/move-command-line-common/src/parser.rs [hunk 2]
```diff
@@ -671,4 +681,25 @@ mod tests {
             );
         }
     }
+
+    fn struct_type_gen() -> impl Strategy<Value = String> {
```
Comment (dariorussi) [inline external-crates/move/crates/move-command-line-common/src/parser.rs:685]: nice

**Review by dariorussi — APPROVED** [review]
thanks

File: external-crates/move/crates/move-command-line-common/src/types.rs [hunk 3]
```diff
@@ -114,7 +114,12 @@ impl Token for TypeToken {
                 let len = 1 + chars.take_while(char::is_ascii_whitespace).count();
                 (Self::Whitespace, len)
             }
-            c if c.is_ascii_alphabetic() => {
+            c if c.is_ascii_alphabetic()
+                || (c == '_'
+                    && chars
+                        .peek()
+                        .is_some_and(|c| identifier::is_valid_identifier_char(*c))) =>
```
Comment (tnowacki) [inline external-crates/move/crates/move-command-line-common/src/types.rs:121]: Maybe, this feels more in line with what my goal was here 
```suggestion
!c.is_digit()
    && chars
        .peek()
        .is_some_and(|c| identifier::is_valid_identifier_char(*c))) =>
```

**Pull request merged**